### ISSUE TITLE
Block ads-api.x.com

### ISSUE THREAD

**Issue by celenityy** [badmojr/1Hosts#1782]
<br>
<h2 align="center">Submit Form</h2>
<div align="center">
  Get your issue resolved quickly! Fill in the form accurately.
</div>

**Domains:**
```
ads-api.x.com
```
<br>

**Details:**

As the name suggests, appears to be a new subdomain related to advertising/tracking on X/Twitter. Noticed it earlier when using Twitter, I blocked it and confirmed it hasn't caused any issues.

<div align="center">
  <h2>Thank you for supporting 1Hosts.</h2>
  <sub>It’s people like you who make these lists great! ❤<ub>
</div>

**Comment by spirillen** [comment]
> I blocked it and confirmed it hasn't caused any issues.

Could you confirm it removed ads/tracking or is it a simple game of guessing?

**Comment by spirillen** [comment]
it's the `cname` for s.twitter.com but it is blacklisted by [twtrdns.net](https://github.com/mypdns/matrix/issues/475)

```
drill -T ads-api.twitter.com
.       518400  IN      NS      a.root-servers.net.
.       518400  IN      NS      b.root-servers.net.
.       518400  IN      NS      c.root-servers.net.
.       518400  IN      NS      d.root-servers.net.
.       518400  IN      NS      e.root-servers.net.
.       518400  IN      NS      f.root-servers.net.
.       518400  IN      NS      g.root-servers.net.
.       518400  IN      NS      h.root-servers.net.
.       518400  IN      NS      i.root-servers.net.
.       518400  IN      NS      j.root-servers.net.
.       518400  IN      NS      k.root-servers.net.
.       518400  IN      NS      l.root-servers.net.
.       518400  IN      NS      m.root-servers.net.
com.    172800  IN      NS      f.gtld-servers.net.
com.    172800  IN      NS      d.gtld-servers.net.
com.    172800  IN      NS      a.gtld-servers.net.
com.    172800  IN      NS      b.gtld-servers.net.
com.    172800  IN      NS      j.gtld-servers.net.
com.    172800  IN      NS      h.gtld-servers.net.
com.    172800  IN      NS      g.gtld-servers.net.
com.    172800  IN      NS      e.gtld-servers.net.
com.    172800  IN      NS      i.gtld-servers.net.
com.    172800  IN      NS      l.gtld-servers.net.
com.    172800  IN      NS      m.gtld-servers.net.
com.    172800  IN      NS      c.gtld-servers.net.
com.    172800  IN      NS      k.gtld-servers.net.
twitter.com.    172800  IN      NS      a.r06.twtrdns.net.
twitter.com.    172800  IN      NS      b.r06.twtrdns.net.
twitter.com.    172800  IN      NS      c.r06.twtrdns.net.
twitter.com.    172800  IN      NS      d.r06.twtrdns.net.
twitter.com.    172800  IN      NS      b.u06.twtrdns.net.
twitter.com.    172800  IN      NS      a.u06.twtrdns.net.
twitter.com.    172800  IN      NS      c.u06.twtrdns.net.
twitter.com.    172800  IN      NS      d.u06.twtrdns.net.
ads-api.twitter.com.    293     IN      CNAME   s.twitter.com.
s.twitter.com.  900     IN      A       104.244.42.3
twitter.com.    13999   IN      NS      a.r06.twtrdns.net.
twitter.com.    13999   IN      NS      a.u06.twtrdns.net.
twitter.com.    13999   IN      NS      b.r06.twtrdns.net.
twitter.com.    13999   IN      NS      b.u06.twtrdns.net.
twitter.com.    13999   IN      NS      c.r06.twtrdns.net.
twitter.com.    13999   IN      NS      c.u06.twtrdns.net.
twitter.com.    13999   IN      NS      d.r06.twtrdns.net.
twitter.com.    13999   IN      NS      d.u06.twtrdns.net.
```

**Comment by celenityy** [comment]
> The Twitter Ads API is a legitimate tool for developers and advertisers to manage ad campaigns programmatically—think creating ads, targeting audiences, or pulling analytics. The API lives under Twitter’s infrastructure (ads-api.twitter.com).

To confirm: 1Hosts considers “targeting audiences” and “pulling analytics” as legitimate functionality?

I’d also like to emphasize this domain is regularly called for all Twitter users, regardless of whether they create ads.

**Comment by b4dmojr** [comment]
What do you think blocking it accomplishes?

**Comment by b4dmojr** [comment]
Its the same case with ads.google.com.
It's a tool for developers and advertisers. Doesn't necessarily serve ads, nor analytics.

**Comment by celenityy** [comment]
> Its the same case with ads.google.com. It's a tool for developers and advertisers. 

I have no idea why you wouldn't block the primary domain associated with the most prolific ad surveillance service (`ads.google.com`), but that's probably out of scope for this issue. Neither `ads.google.com` or `ads-api.x.com` are tools for developers, these 'services' are specifically designed for advertisers to track users.

> Doesn't necessarily serve ads, nor analytics.

`ads-api.x.com` is indeed used for analytics/tracking. For instance, I have noticed calls to:

`https://ads-api.x.com/12/measurement/dcm_local_id`

**Comment by b4dmojr** [comment]
A search reveals that this has long been blocked on Pro, picked up from upstream.
I assume you want it blocked on Lite?

https://rethinkdns.com/search?q=ads-api.x.com

**Comment by celenityy** [comment]
> A search reveals that this has long been blocked on Pro, picked up from upstream.

I don't believe it was ~9 months ago when I filed this issue...

> I assume you want it blocked on Lite?

Sure; it's actively used for tracking and has never been known to cause breakage, so I'd block it there as well; but do what you feel is best.
